File: Array/leetcode128.java
import java.util.HashSet;

class leetcode128 {
    public int longestConsecutive(int[] nums) {

        int result_length =0;
        int max_length=0;

        //해시셋 - 중복된 숫자 찾기 및 탐색에 수월
        HashSet<Integer> hashset = new HashSet<>();

        for(int num: nums){
            hashset.add(num);
        }

        for(int num: hashset){
            if(!hashset.contains(num-1)){
                int current_num = num;
                result_length= 1;

                while(hashset.contains(current_num+1)){
                    current_num++;
                    result_length++;

                }

              max_length = Math.max(max_length, result_length);
                
            }
        }
    return max_length;
        
    }
}

File: Array/leetcode128.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:

        check = set()
        max_length = 0
        length = 0

        for i in nums :
            check.add(i)

        for num in check:
            if num - 1 not in check:
                length = 1
                current_num = num

                while current_num +1 in check:
                    current_num +=1
                    length +=1 

            max_length = max(max_length , length)

        return max_length

                
            
            